crates/acir/src/fallback/mod.rs

```diff
@@ -0,0 +1,93 @@
+use noir_field::FieldElement;
+
+use crate::{
+    circuit::{gate::Directive, Gate},
+    native_types::{Expression, Witness},
+};
+
+// Perform bit decomposition on the provided expression
+pub fn split(
+    gate: Expression,
+    bit_size: u32,
+    mut num_witness: u32,
+    new_gates: &mut Vec<Gate>,
+) -> Vec<Witness> {
+    let mut w_result = Vec::new(); //bit vector
+    let mut g = gate.clone();
+    let mut two_pow = FieldElement::one();
+    let mut intermediate_gates = Vec::new();
+    for _i in 0..bit_size {
+        let w = Witness(num_witness);
+        num_witness += 1;
+        w_result.push(w);
+        let w_bin = Expression {
+            mul_terms: vec![(FieldElement::one(), w, w)],
+            linear_combinations: vec![(-FieldElement::one(), w)],
+            q_c: FieldElement::zero(),
+        };
+        intermediate_gates.push(Gate::Arithmetic(w_bin));
+        g.linear_combinations.push((-two_pow, w));
+        two_pow = FieldElement::from(2_i128) * two_pow;
+    }
+    new_gates.push(Gate::Directive(Directive::Split { a: gate, b: w_result.clone(), bit_size }));
+    new_gates.extend(intermediate_gates);
+    g.sort();
```

Comment:
why do you need to sort here?

Reply:
Gates need to be sorted at several places, for the solver and when expressions are multiplied in acir_gen, so I prefer to sort in case of. It's easy and safe.

Reply:
This seems like bad practice and or a problem with the way the IR is being handled; the compiler shouldn't need to sort gates in the IR.

Reply:
why not? It makes some computations simpler to do.

Reply:
@kevaundray I resolve the conversation, if you still think it should not be sorted, I can create a pr and remove the line (the code will still work) or you can also create an issue.